client/app/scripts/liveblog-edit/module.js

```diff
@@ -312,7 +312,6 @@ define([
                 modal.confirm(gettext('Are you sure you want to remove the blog image?')).then(function() {
                     deregisterPreventer();
                     vm.newBlog.picture_url = null;
-                    vm.newBlog.picture = null;
```

Comment:
If we update with null, then is not possible to pull blogs with `picture` object via embed.
And a PUT method to actually remove that property it will cause different issues ( I think you had those issues ).

Also this solution is actually used form superdesk in `avatar` removal process.